src/cls/tabular/fbwriter_skyroot.cc

```diff
@@ -0,0 +1,784 @@
+        // ---- >>>>
+        std::vector< std::string > an_index_pair ;
+        an_index_pair.push_back( key ) ;
+        an_index_pair.push_back( std::to_string( filedata.string_vect_cnt ) ) ;
+        filedata.indexer.push_back( an_index_pair ) ;
+        // <<<< ----
+        std::vector< std::string > an_empty_vect_strs ;
+        std::vector< flatbuffers::Offset<flatbuffers::String> > an_empty_vect ;
+        filedata.listof_string_vect_raw_strs.push_back( an_empty_vect_strs ) ;
+        filedata.listof_string_vect_raw.push_back( an_empty_vect ) ;
+        filedata.string_vect_cnt++ ;
+      }
+      else {
+        std::cout << ">>2 unrecognized schema_datatype '" << this_dt << "'" << std::endl ;
+        exit( 1 ) ;
+      }
+    } //for loop
+
+    // read csv (pipe delimited) from file
+    std::ifstream infile( inputs.filename ) ;
+    std::string line ;
+    int itnum = 0 ; //need this for some reason?
+    while ( infile.good() && itnum < nrows ) {
+      getline ( infile, line ) ;
+      std::vector< std::string > data_vect = parse_psv_str( line ) ;
+      for( int i = 0; i < data_vect.size(); i++ ) {
+        int attnum = i ;
+        std::string atttype = schema_datatypes[i] ;
+        std::string attkey  = "att" + std::to_string( attnum ) + "-" + atttype ;
+        uint64_t att_vect_id = filedata.get_index( attkey ) ;
+        std::string datum = data_vect[i] ;
+        std::cout << attkey << "," << att_vect_id <<  "," << datum << ";" ;
+
+        // int
+        if( atttype == "int" ) {
+          filedata.listof_int_vect_raw[ att_vect_id ].push_back( std::stoi( datum ) ) ;
+        }
+        // float
+        else if( atttype == "float" ) {
+          filedata.listof_float_vect_raw[ att_vect_id ].push_back( std::stof( datum ) ) ;
+        }
+        // string
+        else if( atttype == "string" ) {
+          filedata.listof_string_vect_raw_strs[ att_vect_id ].push_back( datum ) ;
+          filedata.listof_string_vect_raw[ att_vect_id ].push_back( builder.CreateString( datum ) ) ;
+        }
+        // oops
+        else {
+          std::cout << ">>3 unrecognized atttype '" << atttype << "'" << std::endl ;
+          exit( 1 ) ;
+        }
+      }
+      std::cout << std::endl ;
+      itnum++ ;
+    }
+    infile.close();
+
+    filedata.toString() ;
+
+    // --------------------------------------------- //
+    // build out Rows flatbuffer
+    // --------------------------------------------- //
+    std::vector< flatbuffers::Offset< Tables::Record > > row_records ;
+
+    // create Records as lists of Datas
+    for( unsigned int i = 0; i < nrows; i++ ) {
+
+      // record data
+      std::vector< flatbuffers::Offset< void > > data_vect ;
+
+      for( unsigned int j = 0; j < filedata.indexer.size(); j++ ) {
+
+        auto key   = filedata.indexer[j][0] ;
+        auto index = filedata.indexer[j][1] ;
+
+        if( boost::algorithm::ends_with( key, "-int" ) ) {
```

Comment:
can you please add a comment block here to explain, i'm not entirely sure what this section is doing, thanks.